File: app/src/main/java/isden/mois/magellanlauncher/holders/HistoryDetail.java
package isden.mois.magellanlauncher.holders;

import isden.mois.magellanlauncher.IsdenTools;

/**
 * Created by isden on 16.08.15.
 */
public class HistoryDetail {
    public String date;
    public long spent;
    public double speed;
    public long timestamp;

    public HistoryDetail(String date, long timestamp, long spent) {
        this.date = date;
        this.timestamp = timestamp;
        this.spent = spent;
    }

    public String getDate() {
        if (date == null) {
            return "";
        }
        return this.date;
    }

    public String getSpent() {
        return IsdenTools.prettyTime(this.spent);
    }
}

File: app/src/main/java/isden/mois/magellanlauncher/models/HistoryDetail.kt
package isden.mois.magellanlauncher.models

import isden.mois.magellanlauncher.utils.formatHumanTime

class HistoryDetail(
        var date: String,
        var timestamp: Long,
        var spent: Long,
        var progress: Int,
        var pages: Int,
        var speed: Int
) {
    fun spentTime(): String {
        return formatHumanTime(spent)
    }

    fun formatSpeed(): String = (60 * 60 * 1000 / speed).toString()
}